yaml/yaml_test.go

```diff
@@ -113,3 +113,38 @@ func TestArray(t *testing.T) {
 	assert.Equal(t, verify[0]["c"], 3)
 	assert.Equal(t, verify[1]["c"], 4)
 }
+
+type hasVars struct {
+	Vars []variable
+}
+
+type variable struct {
+	Default interface{}
+}
```

Comment:
these types are unique to the test case only. Consider moving the type declaration into TestStructEmptyArray

Reply:
Done.